compiler/ilgen/IlBuilder.cpp

```diff
@@ -770,6 +770,34 @@ IlBuilder::VectorLoadAt(TR::IlType *dt, TR::IlValue *address)
    return returnValue;
    }
 
+/**
+ * @brief IlBuilder::StructFieldAddress IlGen to get the address of a struct field
+ * @param structName is the name of the struct type the field belongs to
+ * @param fieldName is the name of the field itself
+ * @param obj is an IlValue representing the instance (address) of the struct
+ * @return IlValue representing the address of the field in the struct
+ */
+TR::IlValue *
+IlBuilder::StructFieldAddress(const char* structName, const char* fieldName, TR::IlValue* obj) {
+   auto offset = typeDictionary()->OffsetOf(structName, fieldName);
+   auto ptype = typeDictionary()->PointerTo(typeDictionary()->GetFieldType(structName, fieldName));
+   auto addr = Add(obj, ConstInt64(offset));
```

Comment:
I see your point.

I think I understand how your second approach works in Trees, but I'm not sure how it maps to existing JitBuilder abstractions.

Currently, TR::IlValue cannot represent a struct "obj", only a pointer to an obj. This makes it impossible to load a field that is itself a struct. The purpose of this API is to load the address of the field so that an indirect load/store can then be done.

Admittedly, this is a dirty hack and I should document it is as such. I'm only contributing this change as I've found some use for it in my own project and it's general enough to be useful in other projects as well. Perhaps @mstoodle should have a say in this. @charliegracie might also have some thoughts.